hypothesis-python/src/hypothesis/extra/cli.py

```diff
@@ -0,0 +1,140 @@
+        try:
+            return importlib.import_module(s)
+        except ImportError:
+            pass
+        if "." not in s:
+            modulename, module, funcname = "builtins", builtins, s
+        else:
+            modulename, funcname = s.rsplit(".", 1)
+            try:
+                module = importlib.import_module(modulename)
+            except ImportError:
+                raise click.UsageError(
+                    f"Failed to import the {modulename} module for introspection.  "
+                    "Check spelling and your Python import path, or use the Python API?"
+                )
+        try:
+            return getattr(module, funcname)
+        except AttributeError:
+            public_names = [name for name in vars(module) if not name.startswith("_")]
+            matches = get_close_matches(funcname, public_names)
+            raise click.UsageError(
+                f"Found the {modulename!r} module, but it doesn't have a "
+                f"{funcname!r} attribute."
+                + (f"  Closest matches: {matches!r}" if matches else "")
+            )
+
+    @main.command()  # type: ignore  # Click adds the .command attribute
+    @click.argument("func", type=obj_name, required=True, nargs=-1)
+    @click.option("--idempotent", "writer", flag_value="idempotent")
+    @click.option("--equivalent", "writer", flag_value="equivalent")
+    @click.option("--roundtrip", "writer", flag_value="roundtrip")
+    @click.option(
+        "--style",
+        type=click.Choice(["pytest", "unittest"]),
+        default="pytest",
+        help="pytest-style function, or unittest-style method?",
+    )
+    @click.option(
+        "-e",
+        "--except",
+        "except_",
+        type=obj_name,
+        multiple=True,
+        help="dotted name of exception(s) to ignore",
+    )
+    def write(func, writer, except_, style):  # noqa: D301  # \b disables autowrap
+        """`hypothesis write` writes property-based tests for you!
+
+        Type annotations are helpful but not required for our advanced introspection
+        and templating logic.  Try running the examples below to see how it works:
+
+        \b
+            hypothesis write gzip
+            hypothesis write re.compile --except re.error
+            hypothesis write --style=unittest --idempotent sorted
+            hypothesis write --equivalent ast.literal_eval eval
+            hypothesis write --roundtrip json.dumps json.loads
+        """
+        # NOTE: if you want to call this function from Python, look instead at the
+        # ``hypothesis.extra.ghostwriter`` module.  Click-decorated functions have
+        # a different calling convention, and raise SystemExit instead of returning.
+        if writer is None:
+            writer = "magic"
+        elif writer == "idempotent" and len(func) > 1:
+            raise click.UsageError("Test functions for idempotence one at a time.")
```

Comment:
```suggestion
            raise click.UsageError("A test for idempotence can only accommodate one function at a time")
```

Reply:
I think I prefer the current phrasing, which explains what to do directly.

(and as a general statement the latter is not exactly correct, though it might be good style.  We might also upgrade the idempotence ghostwriter later)